app/src/main/java/org/systers/mentorship/viewmodels/ResetPasswordViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package org.systers.mentorship.viewmodels
+
+import android.annotation.SuppressLint
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.observers.DisposableObserver
+import io.reactivex.schedulers.Schedulers
+import org.systers.mentorship.MentorshipApplication
+import org.systers.mentorship.R
+import org.systers.mentorship.remote.datamanager.UserDataManager
+import org.systers.mentorship.remote.requests.ResetPassword
+import org.systers.mentorship.remote.responses.CustomResponse
+import org.systers.mentorship.utils.CommonUtils
+import retrofit2.HttpException
+import java.io.IOException
+import java.util.concurrent.TimeoutException
+
+/**
+ * This class represents the [ViewModel] used for ResetPasswordFragment
+ */
+class ResetPasswordViewModel : ViewModel() {
+
+    private val userDataManager: UserDataManager = UserDataManager()
+    val successfulUpdate: MutableLiveData<Boolean> = MutableLiveData()
+    lateinit var message: String
+
+    /**
+     * update the password of the current user
+     */
+    @SuppressLint("CheckResult")
+    fun changePassword(resetPassword: ResetPassword) {
+        userDataManager.updatePassword(resetPassword)
+                .subscribeOn(Schedulers.newThread())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeWith(object : DisposableObserver<CustomResponse>() {
+                    override fun onNext(t: CustomResponse) {
```

Comment:
```suggestion
                    override fun onNext(customResponse: CustomResponse) {
```
Why t? I would understand if it was r. I'm just curious